tests/attention/test_mla_dispatch.py

```
@@ -0,0 +1,446 @@
+
+    with pytest.raises(ValueError, match="requires both kv_len and page_table"):
+        wrapper.run(
+            query=torch.empty(1, 1, 576),
+            kv_cache=torch.empty(1, 1, 576),
+        )
+
+
+def test_independent_split_to_packed_is_rejected_without_copy():
+    left, right = torch.empty(1, 1, 2), torch.empty(1, 1, 3)
+
+    with pytest.raises(ValueError, match="query.*packed representation zero-copy"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)